api/v1beta1/kuadrant_types.go

```diff
@@ -79,12 +86,25 @@ type KuadrantSpec struct {
 	// will add the configuration required to enable mTLS between an Istio provided
 	// gateway and the Kuadrant components.
 	MTLS *MTLS `json:"mtls,omitempty"`
+	// +optional
+	// Components configures optional Kuadrant components
+	Components Components `json:"components,omitempty"`
```

Comment:
Consider using a pointer to omit the field when it is empty. Embedded struct do not have empty value, hence `omitempty` is useless. In etcd components and developerportal fields will be stored. 

```yaml
components:
  developerPortal: {}
observability: {}
```